### PR TITLE
Add table functions range and generate_series for TIMESTAMPTZ

### PR DESCRIPTION
**Pull request by papparapa**
This PR adds `range` and `generate_series` table functions for `TIMESTAMPTZ` in ICU extension.

```
range(start: TIMESTAMPTZ, stop: TIMESTAMPTZ, step: INTERVAL)
generate_series(start: TIMESTAMPTZ, stop: TIMESTAMPTZ, step: INTERVAL)
```
As always, for the `range` function, the `stop` parameter is exclusive, while for `generate_series`, it is inclusive.

```sql
D SET TimeZone = 'America/Los_Angeles';
D SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ'1991-06-01 00:00:00-07', INTERVAL '1 MONTH ago') tbl(d);
┌──────────────────────────┐
│            d             │
│ timestamp with time zone │
├──────────────────────────┤
│ 1992-01-01 00:00:00-08   │
│ 1991-12-01 00:00:00-08   │
│ 1991-11-01 00:00:00-08   │
│ 1991-10-01 00:00:00-07   │
│ 1991-09-01 00:00:00-07   │
│ 1991-08-01 00:00:00-07   │
│ 1991-07-01 00:00:00-07   │
└──────────────────────────┘
D SELECT d FROM generate_series(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1991-06-01 00:00:00-07', -INTERVAL '1 MONTH') tbl(d);
┌──────────────────────────┐
│            d             │
│ timestamp with time zone │
├──────────────────────────┤
│ 1992-01-01 00:00:00-08   │
│ 1991-12-01 00:00:00-08   │
│ 1991-11-01 00:00:00-08   │
│ 1991-10-01 00:00:00-07   │
│ 1991-09-01 00:00:00-07   │
│ 1991-08-01 00:00:00-07   │
│ 1991-07-01 00:00:00-07   │
│ 1991-06-01 00:00:00-07   │
└──────────────────────────┘
```

### CODE REVIEW

File: test/sql/table_function/icu_range_timestamptz.test [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+# start is bigger than end but we have a positive interval
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1993-01-01 00:00:00-08', TIMESTAMPTZ '1992-01-01 00:00:00-08', INTERVAL '1 MONTH') tbl(d)
+
+# composite interval with negative types not supported
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1992-12-31 12:00:00-08', INTERVAL '1 MONTH' - INTERVAL '1 HOUR') tbl(d)
+
+# Infinities will overflow or cause infinite loops (PG behaviour!) so we ban them
```
Comment (hawkfish) [inline test/sql/table_function/icu_range_timestamptz.test:133]: Heh PG acknowledged the bug for the last release and fixed it. I have a medallion to prove it 😄

**Review by hawkfish — APPROVED** [review]
Nice work! And thanks for adding table functions to the extension API - that is really powerful.

**Comment by Mytherin** [comment]
Thanks!

File: test/sql/table_function/icu_range_timestamptz.test (hunk 1 above)
Comment (papparapa) [reply test/sql/table_function/icu_range_timestamptz.test:133]: That's great!
Finding a bug in edge cases seems a good way for me to get the PG medallion 🤔

**Pull request merged**